app/src/main/java/com/example/vaccinespotter/worker/RetrofitManager.java

```diff
@@ -0,0 +1,57 @@
+package com.example.vaccinespotter.worker;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.example.vaccinespotter.notifications.VaccineNotificationManager;
+import com.example.vaccinespotter.apiinterface.VaccineSlots;
+import com.example.vaccinespotter.models.Center;
+import com.example.vaccinespotter.models.Centers;
+import com.example.vaccinespotter.models.NotificationModel;
+import com.example.vaccinespotter.models.Session;
+import java.util.List;
+import retrofit2.Call;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitManager {
+    private final String TAG = "RetrofitManager";
+    private static final int age18 = 18;
+    private static final int districtId = 312;
+    private final Retrofit retrofit = new Retrofit.Builder()
+            .baseUrl("https://cdn-api.co-vin.in")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build();
+
+    private final VaccineSlots vaccineSlots = retrofit.create(VaccineSlots.class);
+
+    public boolean queryCowin(VaccineNotificationManager notificationManager, Context context, String date) {

```

Comment:
Better to not have this dependency of NotificationManager here and return the list of valid NotificationModel.
You can then notify user from the worker itself.